corehq/apps/export/models/new.py

```diff
@@ -2470,6 +2477,48 @@ def user_to_be_notified(cls, domain, user):
             )
         )
 
+
+class DataFile(models.Model):
+    domain = models.CharField(max_length=126, db_index=True)
+    filename = models.CharField(max_length=255)
+    description = models.CharField(max_length=255)
+    content_type = models.CharField(max_length=255)
+    blob_id = models.CharField(max_length=255)
+    content_length = models.IntegerField(null=True)
+
+    class Meta(object):
+        app_label = 'export'
+
+    def get_blob(self):
+        db = get_blob_db()
+        try:
+            blob = db.get(self.blob_id)
+        except (KeyError, NotFound) as err:
+            raise NotFound(str(err))
+        return blob
+
+    def save_blob(self, uploaded_file):
+        content_string = StringIO()
+        for chunk in uploaded_file.chunks():
+            content_string.write(chunk)
+        self.content_type = magic.from_buffer(content_string.read(1024))
```

Comment:
I also wanted to mention why I don't validate the file type against a set of acceptable document types. When I looked at some of the possible values of `content_type` I realised how fine-grained it can be -- like the file type given for a "docx" file created by LibreOffice is different from the file type of a "docx" file created by MS Office. I realised it was going to be a support issue if we didn't have all possible permutations, and getting that list might be hard. So I went with Ben's suggestion and just limited the file size.

(It gets even trickier when it comes to zip files, because you have to choose not to accept zip files, which users might not like, or to accept all zip files, so you can effectively upload anything you want (so why bother checking content types at all (this is the option WhatsApp went with, btw)) ... or you inspect inside unencrypted zip files, in which case you're doubling the number of acceptable content types ... and then what do you do with encrypted zip files (which is what a password-protected docx file is)? ... A rabbit hole I didn't to fall down.)